### ISSUE TITLE
OpentelemetryReq: log when path_params not set and continue

### ISSUE THREAD

**Issue by derekkraan** [open-telemetry/opentelemetry-erlang-contrib#351]
Continuing from a discussion I had on Slack with @bryannaegele.

This pertains to the behaviour present in the OpentelemetryReq library, where the request can be halted under the following circumstances:
- `:no_path_params` option is unset
- `path_params` has not been used in the `Req.Request`

The main issue from my point of view is that the instrumentation is changing application behaviour by halting the request [here](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_req/lib/opentelemetry_req.ex#L216-L222).

**Describe the solution you'd like**
- remove the `:no_path_params` option in favour of the branch `no_path_params: true`.
- In other words, fall back from path_params `HTTP GET /foo/:bar` to simply `HTTP GET`
- Log (Bryan suggested to `info`, I'd suggest `debug`) when this fallback occurs.
- _Never_ halt a request.

**Describe alternatives you've considered**
None

**Additional context**
None
